client/my-sites/stats/site.jsx

```diff
@@ -192,6 +201,31 @@ class StatsSite extends Component {
 		// For the new date picker
 		const isDateControlEnabled = config.isEnabled( 'stats/date-control' );
 
+		// Set up a custom range for the chart.
+		// Dependant on new date range picker controls.
+		let customChartRange = null;
+		if ( isDateControlEnabled ) {
+			// Sort out end date for chart.
+			const chartEnd = this.getValidDateOrNullFromInput( context.query?.chartEnd );
+			if ( chartEnd ) {
+				customChartRange = { chartEnd };
+			} else {
+				customChartRange = { chartEnd: moment().format( 'YYYY-MM-DD' ) };
+			}
+			// Sort out quantity for chart.
+			// ToDo: Update to take period into account.
+			const chartStart = this.getValidDateOrNullFromInput( context.query?.chartStart );
+			const isSameOrBefore = moment( chartStart ).isSameOrBefore( moment( chartEnd ) );
+			if ( chartStart && isSameOrBefore ) {
+				const diff = moment( chartEnd ).diff( moment( chartStart ), 'days' );
+				// Make sure quantity includes start date.
+				this.state.customChartQuantity = diff + 1;
```

Comment:
would it be better to use `this.setState()` here instead of mutating the state directly?

Reply:
I don't think we want it to trigger another re-rendering here. Otherwise, it'll end up with indefinite re-rendering. Correct me if I'm wrong @a8ck3n , the approach here merely overrides the original value without triggering re-rending I think.

Reply:
Ahh.. gotcha. That makes sense, I think you're probably right about that Jasper.